README.md

```diff
@@ -0,0 +1,209 @@
+- **OrderController**: Manages order processing, including order creation, updating, and viewing order history.
+- **CategoryController**: Manages category processing, including category creation, updating, and viewing.
+- **ShoppingCartController**: Manages shopping cart processing, including creation, updating, and viewing items (books and other products).
+
+### Services
+
+- **OrderService**: Manages the order processing logic, including placing orders in the database, retrieving order history, and calculating the total price of all items.
+- **BookService**: Manages the book processing logic, including book management for admins and book retrieval for users.
+- **CategoryService**: Manages category processing logic, including category management for admins and category retrieval for users.
+- **ShoppingCartService**: Handles the business logic related to shopping carts, including adding items and updating the cart.
+- **UserService**: Contains business logic for user operations such as registration and user retrieval by username (email).
+
+### Repositories
+
+- **BookRepository**: Interface for CRUD operations on Book entities and special methods for finding books by category.
+- **ShoppingCartRepository**: Interface for CRUD operations on ShoppingCart entities and methods for identifying a shopping cart by user ID.
+- **CartItemRepository**: Interface for CRUD operations on CartItem entities.
+- **OrderRepository**: Interface for CRUD operations on Order entities.
+- **OrderItemRepository**: Interface for CRUD operations on OrderItem entities and methods for retrieving order items by order ID.
+- **UserRepository**: Interface for CRUD operations on User entities and methods for checking if the user is already registered.
+- **RoleRepository**: Interface for CRUD operations on Role entities, including methods for identifying user roles.
+
+## Setup and Usage
+
+To get started with this project, follow these steps:
+
+1. **Clone the repository**:
+    ```sh
+    git clone https://github.com/bodya4243/online-book-store.git
+    cd online-book-store
+    ```
+
+2. **Build the project using Maven**:
+    ```sh
+    mvn clean install
+    ```
+
+3. **Run the application**:
+    ```sh
+    mvn spring-boot:run
+    ```
+   Technologies and Tools
+   Key Technologies
+   Java Development Kit (JDK) 17
+   Maven
+   MySQL
+   Spring Boot
+   H2 Database (for testing)
+   Lombok
+   MapStruct
+   Liquibase
+   SpringDoc OpenAPI
+   JWT (JSON Web Token)
+   Docker Compose
+   
+
+
+4. **Access the application**:
+   - The application runs on: http://localhost:8080.
+   - API documentation is available at: http://localhost:8080/swagger-ui.html.
+
+### Docker Setup
+
+**Swagger**:
+   - You can use Swagger for manipulating API endpoints.
+
+**Docker**:
+   - You can download the container from Docker and start the API.
+
+# MySQL environment variables
```

Comment:
There are not only MySQL variables
```suggestion
# Environment variables
```

Reply:
also if these variables are required for docker and for usual app run - it should be outside the docker section - if it is a sub-topic of the docker run guide - it should be of a smaller size(`###`) than it's parent (`Docker setup`) topic